dbms/src/TiDB/Schema/tests/gtest_table_info.cpp

```
@@ -0,0 +1,435 @@
+        "index_info":[{"backfill_state":0,"comment":"","id":1,"idx_cols":[{"length":-1,"name":{"L":"vec","O":"vec"},"offset":1}],"idx_name":{"L":"v","O":"v"},"index_type":99,"is_global":false,"is_invisible":false,"is_primary":false,"is_unique":false,"mv_index":false,"state":3,"tbl_name":{"L":"","O":""},"vector_index":{"dimension":3,"distance_metric":"COSINE"}}],
+        "is_columnar":false,"is_common_handle":false,"max_col_id":2,"max_cst_id":0,"max_fk_id":0,"max_idx_id":1,"max_shard_row_id_bits":0,"name":{"L":"t","O":"t"},"partition":null,"pk_is_handle":true,"revision":5,"sequence":null,"state":5,"stats_options":null,"temp_table_type":0,"update_timestamp":452784611061923843,"version":5,"view":null})json",
+    };
+
+    for (const auto & c : failure_case)
+    {
+        ASSERT_THROW({ TableInfo table_info(c, NullspaceID); }, DB::Exception) << c;
+    }
+}
+CATCH
+
+struct StmtCase
```

Clang (14.0.6)

[error] constructor does not initialize these fields: table_or_partition_id, tombstone, db_info_json, table_info_json, create_stmt_dm

(cppcoreguidelines-pro-type-member-init,-warnings-as-errors)